### PR TITLE
Allow Reloading of Custom Permissions

### PR DESCRIPTION
**Pull request by willies952002**
Add the ability to reload the custom permissions file by doing "/reload perms"

https://github.com/PaperMC/Paper/issues/49

### CODE REVIEW

File: Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 91bde81..35745ad 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -1018,4 +1018,6 @@ public interface Server extends PluginMessageRecipient {
+     }
+ 
+     Spigot spigot();
++
++    public void reloadPermissions(); // Paper
```
Comment (kashike) [inline Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch:33]: unnecessary `public`

File: Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ 
+     @Override
+     public boolean execute(CommandSender sender, String currentAlias, String[] args) {
++
+         if (!testPermission(sender)) return true;
+ 
++        // Paper Start - Reload permissions.yml (PAPER-49)
++        if ( args.length == 1 ) {
++            if ( args[0].equalsIgnoreCase("permissions") || args[0].equalsIgnoreCase("perms") ) {
++                Command.broadcastCommandMessage(sender, ChatColor.GREEN + "Reloading Server Permissions...");
```
Comment (kashike) [inline Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch:65]: This isn't necessary.

File: Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
++                Bukkit.reloadPermissions();
++                Command.broadcastCommandMessage(sender, ChatColor.GREEN + "Reloaded complete.");
```
Comment (kashike) [inline Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch:67]: `"Permissions successfully reloaded."`

**Comment by kashike** [comment]
This also doesn't work at all, by the way.

**Comment by willies952002** [comment]
it seems that my work is not appreciated

**Comment by vemacs** [comment]
This is not the correct method to reload perms. See: https://github.com/aaomidi/ReloadPermissions/blob/master/src/main/java/co/mccn/reloadpermissions/ReloadPermissions.java#L47

**Comment by willies952002** [comment]
@vemacs I don't want to just use someone else's work as that wouldn't make it my own. I will be keeping this in my own fork until someone feels they actually want it.

File: Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch (hunk 2 above)
Comment (willies952002) [inline Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch:65]: Message Removed

File: Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch (hunk 1 above)
Comment (willies952002) [inline Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch:33]: Removed

File: Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch (hunk 3 above)
Comment (willies952002) [inline Spigot-API-Patches/0026-Add-command-to-reload-permissions.yml.patch:67]: Message Updated

**Comment by willies952002** [comment]
> This also doesn't work at all, by the way.

Are you encountering any errors that would suggest that it's broken? It's currently working fine on my testing server.

**Comment by vemacs** [comment]
Bukkit.reloadPermissions() does not reload permissions.yml. No offense, but I don't know what you're smoking.

**Comment by airhub** [comment]
He's probably smoking some serious Jamacian cocaine.

**Comment by willies952002** [comment]
> Bukkit.reloadPermissions() does not reload permissions.yml. No offense, but I don't know what you're smoking.

That's a new method added that calls server.reloadPermissions()

> He's probably smoking some serious Jamacian cocaine.

I don't smoke anything. Personally, I hate drugs... including prescription ones.

**Comment by willies952002** [comment]
Recreated since I cannot reopen this -> https://github.com/PaperMC/Paper/pull/91

**Pull request closed without merging**